fish-rust/src/env_dispatch.rs

```diff
@@ -0,0 +1,796 @@
+use crate::curses;
+use crate::env::{setenv_lock, unsetenv_lock, EnvMode, EnvStack, Environment};
+use crate::env::{CURSES_INITIALIZED, DEFAULT_READ_BYTE_LIMIT, READ_BYTE_LIMIT, TERM_HAS_XN};
+use crate::ffi::is_interactive_session;
+use crate::flog::FLOGF;
+use crate::output::ColorSupport;
+use crate::wchar::L;
+use crate::wchar::{wstr, WString};
+use crate::wchar_ext::WExt;
+use crate::wutil::fish_wcstoi;
+use crate::wutil::wgettext;
+use std::collections::HashMap;
+use std::ffi::{CStr, CString};
+use std::sync::atomic::{AtomicBool, Ordering};
+
+#[cxx::bridge]
+mod env_dispatch_ffi {
+    extern "Rust" {
+        fn env_dispatch_init_ffi();
+        fn term_supports_setting_title() -> bool;
+        fn use_posix_spawn() -> bool;
+    }
+}
+
+/// List of all locale environment variable names that might trigger (re)initializing of the locale
+/// subsystem. These are only the variables we're possibly interested in.
+#[rustfmt::skip]
+const LOCALE_VARIABLES: &[&wstr] = &[
+    L!("LANG"),       L!("LANGUAGE"), L!("LC_ALL"),
+    L!("LC_COLLATE"), L!("LC_CTYPE"), L!("LC_MESSAGES"),
+    L!("LC_NUMERIC"), L!("LC_TIME"),  L!("LOCPATH"),
+    L!("fish_allow_singlebyte_locale"),
+];
+
+#[rustfmt::skip]
+const CURSES_VARIABLES: &[&wstr] = &[
+    L!("TERM"), L!("TERMINFO"), L!("TERMINFO_DIRS")
+];
+
+/// Whether to use `posix_spawn()` when possible.
+static USE_POSIX_SPAWN: AtomicBool = AtomicBool::new(false);
+
+/// Whether we think we can set the terminal title or not.
+static CAN_SET_TERM_TITLE: AtomicBool = AtomicBool::new(false);
+
+/// The variable dispatch table. This is set at startup and cannot be modified after.
+static VAR_DISPATCH_TABLE: once_cell::sync::Lazy<VarDispatchTable> =
```

Comment:
this no longer leaks. Presumably this is fine

Reply:
Actually, all rust static variables are leaked exactly because of the problems caused by C++ *not* leaking static variables.